### ISSUE TITLE
CommandLineEncoder appears to ignore the height and width parameters when barcode_format is  DATA_MATRIX

### ISSUE THREAD

**Issue by ablegreen** [zxing/zxing#836]
`CommandLineEncoder` appears to ignore the height and width parameters when `barcode_format` is  `DATA_MATRIX`

The following command produces a 14px by 14px image instead of a 500px by 500px image: 

```
java -cp javase/target/javase-3.3.0-jar-with-dependencies.jar com.google.zxing.client.j2se.CommandLineEncoder --barcode_format DATA_MATRIX --height 500 --width 500 abcdefg
```

Here's a `Dockerfile` to reproduce this issue:

```
FROM ubuntu:16.04
RUN apt-get update && apt-get install -y --no-install-recommends default-jdk git maven
RUN git clone --depth 1 --branch zxing-3.3.0 https://github.com/zxing/zxing.git
RUN cd zxing && mvn install -DskipTests
RUN cd zxing/javase && mvn -DskipTests package assembly:single
RUN cd zxing && java -cp javase/target/javase-3.3.0-jar-with-dependencies.jar com.google.zxing.client.j2se.CommandLineEncoder --barcode_format DATA_MATRIX --height 500 --width 500 abcdefg
```

Is this a bug or am I using `CommandLineEncoder` wrong?

**Comment by srowen** [comment]
You're right, I don't believe it actually respects those values. The DataMatrix encoder was adopted from another project and works somewhat differently. If you're interested in looking at plumbing through those values and trying to make it scale up the image and add padding like the QR code writer does I can look at that with you. It's a legitimate issue, yes.

**Comment by axxel** [comment]
Just as a reference for whom it may concern: I recently introduced this missing feature to the new c++ port with this change: https://github.com/nu-book/zxing-cpp/commit/be46af7ddc0da3b955d8547c19d81140a4c707b2

The interesting part is the new generic `Inflate` function that is now used inside the QRWriter, the AztecWriter and the DatamatrixWriter. It takes a `BitMatrix` and inflates it by scaling it up and adding padding as necessary.

**Comment by iphone6fromApple** [comment]
i  on concern

**Comment by iphone6fromApple** [comment]
i on concern

**Comment by Medusa500** [comment]
DATA_MATRIX

**Comment by wvieed** [comment]
V

**Comment by janeneb711** [comment]
I know nothing about source code my last device that was breach is android and now I have an Apple iPhone trying to fix my name and recover hijacked gmail account

**Comment by gredler** [comment]
@srowen Should this be closed? It sounds like it might have been fixed with PR #933.
